recipes/gcc/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+        if self.settings.compiler in ["clang", "apple-clang"]:
+            # xgcc: error: unrecognized command-line option -stdlib=libc++
+            if self.settings.compiler.libcxx == "libc++":
+                self._autotools.cxx_flags.remove("-stdlib=libc++")
+            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
+                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("mpc/1.1.0")
+        self.requires('mpfr/4.0.2')
+        self.requires("gmp/6.1.2")
+        self.requires('zlib/1.2.11')
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
+        if tools.cross_building(self.settings):
+            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "gcc-releases-gcc-%s" % self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _make_args(self):
+        if self.settings.os == "Macos":
+            return ["BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"]
+        return []
+
+    def build(self):
+        # If building on x86_64, change the default directory name for 64-bit libraries to "lib":
+        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
+        tools.replace_in_file(os.path.join(self.source_folder,
+                                           self._source_subfolder, "gcc", "config", "i386", "t-linux64"),
+                              "m64=../lib64", "m64=../lib", strict=False)
+        # Ensure correct install names when linking against libgcc_s;
+        # see discussion in https://github.com/Homebrew/legacy-homebrew/pull/34303
+        tools.replace_in_file(os.path.join(self.source_folder,
+                                           self._source_subfolder, "libgcc", "config", "t-slibgcc-darwin"),
+                              "@shlib_slibdir@", libdir, strict=False)
+        autotools = self._configure_autotools()
+        autotools.make(args=self._make_args)
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def _remove_by_mask(self, pattern):
+        import fnmatch
+        for root, _, filenames in os.walk(self.package_folder):
+            for filename in filenames:
+                if fnmatch.fnmatch(filename, pattern):
+                    os.unlink(os.path.join(self.package_folder, root, filename))
+
+    def package(self):
+        autotools = self._configure_autotools()
+        autotools.install(args=self._make_args)
```

Comment:
I would run `install-strip` instead: https://github.com/docker-library/gcc/blob/master/Dockerfile.template#L125

As you can see, the official GCC docker image uses it. The idea is strip the binaries, so the package size will be much smaller.

Reply:
okay, let's add this as well